File: lua/study/e_module.lua
#!lua 5.3.2
--@guliping
--[[ 使用模块 
http://blog.csdn.net/glp3329/article/details/52713288

上面地址不仅包含了模块所需的知识，也包含了lua元表的知识，
如果希望能更深入，一定要深刻理解lua元表以及元方法。
]]

--引用模块，并赋值到本地局部变量
local eA = require "eA" --后缀名默认  .lua

print(eA)
print("模块内容")
for k,v in pairs(eA) do print(k,v) end

eA.d()
eA.e()


--[[
require 用于搜索 Lua 文件的路径是存放在全局变量 package.path 中，
当 Lua 启动后，会以环境变量 LUA_PATH 的值来初始这个环境变量。
如果没有找到该环境变量，则使用一个编译时定义的默认路径来初始化。
当然，如果没有 LUA_PATH 这个环境变量，也可以自定义设置，
在当前用户根目录下打开 .profile 文件（没有则创建，打开 .bashrc 文件也可以），
例如把 "~/lua/" 路径加入 LUA_PATH 环境变量里：
--]]

--require 引用路径
print(package.path)

--如果找到该文件，那么使用package.loadfile加载文件，
--否则搜索package.cpath的so或者dll文件，通过package.loadlib加载


--C/C++封装的模块

File: py/study/e_module.py
#!python3.4
# @ guliping

from eB import __d  # 但是可以指定引入某个私有成员或者方法
from eB import *
from eB import a
''' 模块
http://blog.csdn.net/glp3329/article/details/52713288
'''

import eA
import sys

print(sys.path)

print(dir(eA))  # 私有的方法和成员不展示
print(eA.__d)  # 虽然是私有的，但是我们可以直接访问

a()
# b()  #未引入，便会报错

b()
# print(__d) #报错，私有成员或者方法，不能通过from xxx import * 引入

print(__d)

# python模块支持在某个目录下添加 __init__.py 作为引入包的初始化函数， __all__ = ["A","B"] 指定需要加载的子模块
